### PR TITLE
CM-49 project initialization

### PR DESCRIPTION
**Pull request by merlindk**

### CODE REVIEW

**Comment by merlindk** [comment]
test this please

**Comment by merlindk** [comment]
test this please

**Comment by merlindk** [comment]
test this please

**Comment by merlindk** [comment]
test this please

**Comment by merlindk** [comment]
test this please

File: src/main/java/com/cema/users/controllers/Controller.java [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+        this.userMapping = userMapping;
+        this.loginService = loginService;
+    }
+
+    @ApiOperation(value = "Retrieve a user data by username", response = User.class)
+    @ApiResponses(value = {
+            @ApiResponse(code = 200, message = "Successfully found user"),
+            @ApiResponse(code = 404, message = "The user you were looking for is not found")
+    })
+    @GetMapping(value = "/users/{username}", produces = {MediaType.APPLICATION_JSON_VALUE})
```
Comment (FranciscodeMaussion) [inline src/main/java/com/cema/users/controllers/Controller.java:55]: Maybe saving /users/ in a constant variable is a good idea...

File: src/main/java/com/cema/users/controllers/Controller.java [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+            @ApiParam(
+                    value = "The username of the user we are looking for.",
+                    example = "merlinds")
+            @PathVariable("username") String userName) {
+
+        LOG.info("Request for user {}", userName);
+
+        CemaUser cemaUser = cemaUserRepository.findCemaUserByUserName(userName);
+        if (cemaUser == null) {
+            throw new UserNotFoundException(String.format("User %s doesn't exits", userName));
```
Comment (FranciscodeMaussion) [inline src/main/java/com/cema/users/controllers/Controller.java]: "User %s doesn't exits" is repeated 3 times, maybe you could use a constant REF http://192.168.100.4:9000/project/issues?id=users-be&open=AXoREALpVXiR6bdbzjHb&resolved=false&sinceLeakPeriod=true&types=CODE_SMELL

File: src/main/java/com/cema/users/domain/User.java [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+import io.swagger.annotations.ApiModelProperty;
+
+import java.util.Date;
+
+public class User {
+
+    @ApiModelProperty(notes = "The username for this user, needed for login", example = "merlinds")
+    private String userName;
+    @ApiModelProperty(notes = "The name of the owner of this user", example = "Merlin")
```
Comment (FranciscodeMaussion) [inline src/main/java/com/cema/users/domain/User.java]: I think we should treat the user as the person itself, so "The name <s>of the owner</s> of this user"
Same applies to the other notes

File: src/main/resources/application.yml [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    activate:
+      on-profile: qa
+
+---
+spring:
+  datasource:
+    url: jdbc:postgresql://192.168.100.6:5432/users
+  config:
+    activate:
+      on-profile: prod
```
Comment (FranciscodeMaussion) [inline src/main/resources/application.yml]: EOF: suggested by github

**Review by FranciscodeMaussion — CHANGES_REQUESTED** [review]

File: src/main/java/com/cema/users/controllers/Controller.java (hunk 1 above)
Comment (merlindk) [reply src/main/java/com/cema/users/controllers/Controller.java:55]: I saw it in your code and I don't like it. but is personal preference, is not something that will change much, and in my opinion it adds clarity by having it like this

File: src/main/java/com/cema/users/controllers/Controller.java (hunk 2 above)
Comment (merlindk) [reply src/main/java/com/cema/users/controllers/Controller.java]: good point

File: src/main/java/com/cema/users/domain/User.java (hunk 3 above)
Comment (merlindk) [reply src/main/java/com/cema/users/domain/User.java]: ok, valid.

File: src/main/resources/application.yml (hunk 4 above)
Comment (merlindk) [reply src/main/resources/application.yml]: ok

**Review by FranciscodeMaussion — APPROVED** [review]

**Comment by merlindk** [comment]
test this please

**Comment by merlindk** [comment]
test this please

**Comment by merlindk** [comment]
test this please

**Comment by merlindk** [comment]
test this please

**Pull request merged**